### DISCUSSION TITLE
Copilot Chat - does it use the GPT-4 language model?

### DISCUSSION THREAD

**Discussion by adamklepacz** [community/community#56396] [Copilot Conversations]
### Select Topic Area

Question

### Body

Hello GitHub Community,

I'm currently trying to deepen my understanding of GitHub's Copilot Chat feature. This impressive tool has undoubtedly made coding much easier and efficient, and its conversation-like interaction with users truly is revolutionary.

As far as I'm aware, previous versions of Copilot made use of models like Codex, which is a descendant of the GPT-3 language model. My question is whether the current version of Copilot Chat utilizes the GPT-4 model? If so, could anyone elaborate on how the application of GPT-4 has further enhanced the effectiveness and precision of the tool?

Any insights on the topic, especially if you could link to relevant resources, would be greatly appreciated. I'm particularly interested in understanding the improvements that GPT-4 might have brought to the Copilot feature in terms of code generation, error prediction, and general interactions.

Thank you in advance for your time and assistance.

Best Regards,
Adam

**Comment by Akash1134** [comment]
Hi @adamklepacz ,

I have responded to a [similar query](https://github.com/orgs/community/discussions/56397). Locking this one as a duplicate. 

Thanks!
